import/index_java/src/org/vufind/index/GeoTools.java

```diff
@@ -106,14 +113,107 @@
                 if (validateCoordinateValues(record, coords)) {
                     String result = String.format("%s %s %s %s", new Object[] {  coords.get('d'),  coords.get('e'),  coords.get('f'),  coords.get('g') });
                     geo_coordinates.add(result);
-                } else {
-                    logger.error(".......... Not indexing INVALID Display coordinates: [ {" + coords.get('d') + "} {" + coords.get('e') + "} {" + coords.get('f') + "} {" + coords.get('g') + "} ]");
                 }
             }
         }
         return geo_coordinates;
     }
 
+    /**
+    * Log coordinate indexing errors to external log file.
+    *
+    * @param  Record record
+    * @param  HashMap coords
+    * @param  String error message
+    */
+   public static void logErrorMessage(Record record, HashMap coords, String message) {
+
+        // Initialize error logging variables
+        String msgError = message;
+        ControlField recID = (ControlField) record.getVariableField("001");
+        String recNum = recID.getData().trim();
+        String coordinates = "Coordinates:  {" + coords.get('d') + "} {" + coords.get('e') + "} {" + coords.get('f') + "} {" + coords.get('g') + "}";
+
+        String logPath = getErrorLogPath();
+        String datePrefix = new SimpleDateFormat("yyyyMMdd_HHmmss").format(new Date());
+        String logFilename = datePrefix + "_CoordinateErrors.txt";
+        String outPath = logPath + "/" + logFilename;
+
+        // Output Error message
+        logger.error("Record ID: " + recNum + " ... " + msgError);
+        logger.error("..... Not indexing INVALID coordinates .... Check coordinate error log at: " + outPath);
+
+        // Log ID and error message and coordinates in error file
+        try {
+             PrintWriter out = new PrintWriter(new BufferedWriter(new FileWriter(outPath, true)));
+             out.println(logFilename + "\t" + recNum + "\t" + msgError + "\t" + coordinates);
+             out.close();
+        } catch (IOException e) {
+             System.out.println("io exception occured");
+             e.printStackTrace();
+        }
+    }
+
+    /**
+    * Get path for coordinate error log file.
+    *
+    * @return String logPath
+    */
+   public static String getErrorLogPath() {
+        String filename = "import.properties";
```

Comment:
I don't think we want to hard-code import.properties here; the properties filename may be overridden as a parameter to the indexer. There may be an easier way to get all of these settings. I don't know the answer off the top of my head, but let me know if you need help digging.

Reply:
I'm going to re-write this section and resubmit.